### PR TITLE
Add clear_progress_for! to user

### PR DESCRIPTION
**Pull request by julian-berbel**
## :dart: Goal
Ease support tasks that require deleting user progress.

## :memo: Details
Adding `clear_progress_for!` to users to allow clearing progress for a user for a specific organization or optionally, all organizations.

## :warning: Dependencies
None.

## :back: Backwards compatibility
Yes.

## :soon: Future work
None.

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -743,4 +743,48 @@
...
+
+    before do
+      organization_1.switch!
+      assignment_1 = exercise_1.submit_solution!(user).tap(&:passed!)
+      create :message, sender: user, assignment: assignment_1, discussion_id: nil
+      organization_2.switch!
+      assignment_2 = exercise_2.submit_solution!(user).tap(&:passed!)
+      create :message, sender: user, assignment: assignment_2, discussion_id: nil
+      create :message, sender: user, assignment: assignment_1, discussion_id: 1
+    end
```
Comment (felipecalvo) [inline spec/models/user_spec.rb:765]: I don't think this last message here can actually occur. Discussion messages don't have an assignment, as far as I know; only teacher-to-student messages do (but then they don't have a discussion id).

In any case the behavior is the same, I think.

**Review by felipecalvo — APPROVED** [review]
Good job! I think you're covering all the cases here, I hope we're not missing anything.

File: spec/models/user_spec.rb (hunk 1 above)
Comment (julian-berbel) [reply spec/models/user_spec.rb:765]: ah, I was not aware of that. I'll do a quick check on it later just in case

Comment (julian-berbel) [reply spec/models/user_spec.rb]: yep, you were absolutely right! as a side note we have 2 messages which have both `assignment: nil` and `discussion_id: nil`... we should probably delete those

Comment (felipecalvo) [reply spec/models/user_spec.rb]: ~We should delete all messages with `discussion:nil` and the whole feature itself~ Yeah, we should

**Pull request merged**